src/backend/sev/builder.rs

```diff
@@ -10,33 +10,75 @@ use crate::backend::kvm::mem::Region;
 
 use std::convert::TryFrom;
 use std::sync::{Arc, RwLock};
+use std::{thread, time};
 
-use anyhow::{Context, Error, Result};
+use anyhow::{Context, Error};
 use kvm_ioctls::Kvm;
 use mmarinus::{perms, Map};
 use primordial::Page;
+use rand::{thread_rng, Rng};
 use sallyport::elf::pf::snp::{CPUID, SECRETS};
 use x86_64::VirtAddr;
 
+const SEV_RETRIES: usize = 3;
+const SEV_RETRY_SLEEP_MS: u64 = 500;
+
 pub struct Builder {
     kvm_fd: Kvm,
     launcher: Launcher<Started, Firmware>,
     regions: Vec<Region>,
     sallyports: Vec<Option<VirtAddr>>,
 }
 
+fn retry<O, F>(func: F) -> anyhow::Result<O>
+where
+    F: Fn() -> anyhow::Result<O>,
```

Comment:
Can't we do this instead?
```suggestion
fn retry<O>(func: impl Fn() -> anyhow::Result<O>) -> anyhow::Result<O>
```

Reply:
I think if we can avoid naming generic type parameters, then we should, especially in cases like this when there's only 1 such parameter

Reply:
done